### PR TITLE
Add new Select component

### PR DESCRIPTION
**Pull request by williamkolmacka**
This Pull Request meets the following criteria:

* [x] Tests have been added/adjusted for my new feature
* [x] New Components are registered in index.js of my project

Add a new Select component. There are still some values in CSS that are not in tokens.

### CODE REVIEW

File: src/Select/Select.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  state = {
+    value: this.props.value,
+  };
+
+  handleChange = (e: SyntheticInputEvent<HTMLSelectElement>) => {
+    this.setState({ value: e.target.value });
+    this.props.onChange(e);
+  };
+
+  renderOption = ({ value, label, visible = true }: Option) => {
```
Comment (wEprous) [inline src/Select/Select.js]: You defined `disabled` in Option type but you are not rendering it anywhere.

File: src/Select/Select.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  &::-ms-expand {
+    display: none;
+  }
+  &:disabled {
+    color: ${({ theme }) => theme.colorTextInputDisabled};
+    background: ${({ theme }) => theme.backgroundInputDisabled};
+    cursor: default;
+  }
+  /* placeholder */
+  &:invalid {
```
Comment (wEprous) [inline src/Select/Select.js]: Please for `placeholder` style use another way. Required attr will probably render native browser error. Do this like:
```js
const isPlaceholder = this.props.placeholder && this.state.value === ""
```
You then pass this const to `StyledSelect` and it'll decide which color should apply.

File: src/Select/Select.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+export const SIZE_OPTIONS = {
+  SMALL: "small",
+  NORMAL: "normal",
+};
+
+type Option = {
+  value: string | number,
+  label?: string,
+  disabled?: boolean,
+  visible?: boolean, // eslint-disable-line react/no-unused-prop-types
```
Comment (wEprous) [inline src/Select/Select.js]: The is no logical reason to use `visible` more, please delete it everywhere.

File: src/Select/Select.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    }
+    return null;
+  };
+
+  render() {
+    const { theme, prefix, label, size, disabled, placeholder, help, error, options } = this.props;
+    const { value } = this.state;
+    return (
+      <React.Fragment>
+        {label && <Text>{label}</Text>}
```
Comment (wEprous) [inline src/Select/Select.js]: I'm completely missing `label` knob in stories. Don't forget that label's color is changing on the selected value.

File: src/Select/__tests__/Select.test.js [hunk 5]
```diff
@@ -14,15 +16,30 @@ const objectOptions = [
   { disabled: true, value: "disabled-six", label: "Disabled Six" },
 ];
 
-const mockChange = jest.fn();
 describe("Select", () => {
   const component = shallow(<Select value="1" options={objectOptions} onChange={mockChange} />);
+  const componentWithPlaceholder = shallow(
```
Comment (wEprous) [inline src/Select/__tests__/Select.test.js]: You can merge these two shallow wrappers into one.

File: src/Select/Select.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+  display: inline-block;
+  font-family: ${({ theme }) => theme.fontFamily};
+  font-size: ${({ theme }) => theme.fontSizeLabelForm};
+  color: ${({ theme, value }) =>
+    value === "" ? theme.colorLabelForm : theme.colorLabelFormFilled};
+  line-height: ${({ theme }) => theme.lineHeightText};
+  margin-bottom: ${({ theme }) => theme.spaceXXSmall};
+`;
+
+const SelectContainer = styled(({ theme, ...props }) => <div {...props} />)`
```
Comment (wEprous) [inline src/Select/Select.js]: Omit `error` or:
```js
const SelectContainer = styled(({ children, className }) => (
  <div className={className}>{children}</div>
))`
```

File: src/Select/Select.js [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+    ${({ theme, error }) => (error ? theme.borderColorInputError : theme.borderColorInput)};
+  border-color: ${({ theme, active, error }) =>
+    active && (error ? theme.borderColorInputError : theme.borderColorInputFocus)};
+  box-shadow: ${({ theme, active, error }) =>
+    active && (error ? theme.boxShadowInputErrorFocus : theme.boxShadowInputFocus)};
+  width: 100%;
+  box-sizing: border-box;
+  cursor: pointer;
+  &:hover {
+    border-color: ${({ theme, active, error }) =>
```
Comment (wEprous) [inline src/Select/Select.js]: Disable `hover` when select is `disabled`

File: src/Select/Select.js [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+
+  render() {
+    const { theme, prefix, label, size, disabled, placeholder, help, error, options } = this.props;
+    const { value, active } = this.state;
+    const isPlaceholder = placeholder && value === "";
+
+    return (
+      <React.Fragment>
+        {label && (
+          <SelectLabel theme={theme} value={value}>
```
Comment (wEprous) [inline src/Select/Select.js]: Should be `select` focused and opened after clicking on `label`?

Comment (williamkolmacka) [reply src/Select/Select.js]: Can make this when we will have custom select ready. For now, it is not so much important.

File: src/Select/Select.js [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+  label?: string,
+  disabled?: boolean,
+};
+
+type Props = {
+  size: $Values<typeof SIZE_OPTIONS>,
+  label?: string,
+  placeholder?: string,
+  value: string,
+  disabled?: boolean,
```
Comment (wEprous) [inline src/Select/Select.js]: When select is disabled, styles are not changed.

File: src/Select/Select.stories.js [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+          },
+        ],
+      },
+    ],
+  }))
+  .addWithChapters("Playground", () => {
+    const placeholder = text("Placeholder", "Select value from list");
+    const size = select("Size", Object.values(SIZE_OPTIONS), SIZE_OPTIONS.NORMAL);
+    const disabled = boolean("Disabled", false);
+    const option = object("Options", objectOptions);
```
Comment (wEprous) [inline src/Select/Select.stories.js]: I would add `value` knob to playground.

File: src/Select/__tests__/Select.test.js [hunk 11]
```diff
@@ -5,6 +5,8 @@ import { shallow } from "enzyme";
 
 import Select from "../Select";
 
+const mockChange = jest.fn();
+const placeholder = "Default placeholder";
 const objectOptions = [
```
Comment (wEprous) [inline src/Select/__tests__/Select.test.js]: `visible` is not needed in this object more.

File: src/Select/Select.js [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+
+  &::-ms-expand {
+    display: none;
+  }
+
+  /* With prefix */
+  padding-left: ${({ prefix }) => prefix && "48px"}; // TODO: Create token
+
+  /* Based on state of select */
+  border-color: ${({ theme, error }) =>
```
Comment (wEprous) [inline src/Select/Select.js]:
```css
border: 1px solid
```
and delete line 44

File: src/Select/Select.js [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+  }
+
+  &:disabled {
+    color: ${({ theme }) => theme.colorTextInputDisabled};
+    background: ${({ theme }) => theme.backgroundInputDisabled};
+    cursor: default;
+  }
+`;
+
+const SelectContainer = styled.div`
```
Comment (wEprous) [inline src/Select/Select.js]: `disabled` attr is render for this `div`

File: src/Select/Select.js [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+  <div className={className}>{children}</div>
+))`
+  display: flex;
+  align-items: center;
+  position: absolute;
+  padding: ${({ theme }) => `0 ${theme.spaceSmall}`};
+  pointer-events: none;
+`;
+
+const SelectSuffix = styled.div`
```
Comment (wEprous) [inline src/Select/Select.js]: `disabled` attr is render for this `div`

File: src/Select/Select.js (hunk 9 above)
Comment (tomashapl) [reply src/Select/Select.js]: Good catch 👏

File: src/Select/Select.js (hunk 12 above)
Comment (tomashapl) [reply src/Select/Select.js]: It defines basic style of border, so I want to let it

File: src/Select/Select.js (hunk 13 above)
Comment (tomashapl) [reply src/Select/Select.js]: Fixed

File: src/Select/__tests__/Select.test.js (hunk 11 above)
Comment (tomashapl) [reply src/Select/__tests__/Select.test.js]: Fixed

File: src/Select/Select.stories.js (hunk 10 above)
Comment (tomashapl) [reply src/Select/Select.stories.js]: Added

File: src/Select/Select.js (hunk 12 above)
Comment (wEprous) [reply src/Select/Select.js]: Ok :)

**Pull request merged**